src/auth-guard/auth-guard.ts

```diff
@@ -40,21 +40,23 @@ export class AngularFireAuthGuard implements CanActivate {
     return this.authState.pipe(
       take(1),
       authPipeFactory(next, state),
-      map(can => typeof can == "boolean" ? can : this.router.createUrlTree(<any[]>can))
+      map(can => typeof can === 'boolean' ? can :
+        Array.isArray(can) ? this.router.createUrlTree(can) : this.router.parseUrl(can)
```

Comment:
Double ternary is kinda hard to read un understand (especially formatted this way).
Consider using a different approach, prob. just few "if" statements